### PR TITLE
Fix #6606: added UI to handle tx different statuses

### LINKED ISSUE #6606: Apply eth_signTransaction UI changes (1.48.x brave-core)
core changes https://github.com/brave/brave-core/pull/16247

Requires UI changes for
1. Post confirmation showing transaction is signed but will be sent by DApp like 
Screen Shot 2022-12-01 at 17 53 49

2. Handle display for new transaction status `Signed`

### LINKED ISSUE #6861: Wallet: Submitted/Completed State after user confirms a transaction
After a user confirms a transaction, add a loading and submitted state
If this user stays at this screen, we will display a completed state when the transaction is completed

Screenshot 2023-02-09 at 4 55 30 PM

Screenshot 2023-02-01 at 11 00 29 AM

216392209-3a4945e1-7e53-4e80-9d38-61df0c6efb28

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes
For each transaction user clicks confirm in the transaction confirmation screen, it will always have UI to handle the changing of the states. It includes when tx is in loading/submitting, tx has been submitted or signed, tx has been confirmed/completed. 

This pull request fixes #6606 #6861

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Test regular transaction status changes:
1. Make either a send a swap transaction via Brave Wallet
2. A transaction confirmation screen will pop up
3. click confirm
4. observe a loading screen will show up 
5. observe after the loading, view will become transaction submitted state which has icon, text and buttons
6. click the ok button 
7. observe it will dismiss the whole screen
8. repeat step 1 to 5
9. click the explorer button
10. observe it will open transaction url 
11. repeat step 1 to 5
12. stay at this screen for few seconds
13. observe the view will change to another state indicates the tx has been completed 
14. click the receipt button
15. observe a transaction detail screen show up
16. close the detail screen and click close button
17. observe the whole screen will be dismissed
18. repeat step 1 to 2
19. click advanced setting in the confirmation screen and edit nonce to be `0`. save it
20. click confirm 
21. observe a loading screen will show up
22. observe after loading, view will become transaction failed with an error message
23. click ok button
24. observe it will dismiss the whole screen.

## Test signed transaction:
1. Setup https://github.com/bbondy/eth-manual-tests
2. Change to any test network and visit http://localhost:8081/request.html
3. setup safara debug console to check any output from http://localhost:8081/request.html
4. Click ethereum.enable and grant permission
5. Fill to address next to eth_signTransaction and click it
6. observe a notification will show up and click the notification to open brave wallet
7. once the wallet is unlocked, a send pending transaction will show up 
8. click confirm 
9. observe a loading state will apear
10. observe after loading, the tx status will change to signed. 
11. click ok to close the screen 
12. check account activities to see the transaction appears with status `Signed`
13. Paste the return signed transaction in console to eth_sendRawTransaction and click
14. Check transactions in account activity and previously `Signed` transaction should be `Confirmed` after a while

## Screenshots:

https://user-images.githubusercontent.com/1187676/219449556-8b7b58e1-d907-4dcb-86f5-782230f484fa.mp4

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+import DesignSystem
+
+struct PendingTransactionView: View {
+  @ObservedObject var confirmationStore: TransactionConfirmationStore
+  let networkStore: NetworkStore
+  let keyringStore: KeyringStore
+  @Binding var isShowingGas: Bool
+  @Binding var isShowingAdvancedSettings: Bool
+  
+  var onClose: () -> Void
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift]: nit: we use `onDismiss` throughout wallet

File: Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift [hunk 2]
```diff
@@ -46,10 +46,14 @@ public class TransactionConfirmationStore: ObservableObject {
...
-      if let tx = transactions.first(where: { $0.id == activeTransactionId }) {
+      if let tx = allTxs.first(where: { $0.id == activeTransactionId }) {
         updateTransaction(with: tx)
-      } else if let firstTx = transactions.first {
+        activeTxStatus = tx.txStatus
+      } else if let firstTx = unapprovedTxs.first {
         updateTransaction(with: firstTx)
+        activeTxStatus = firstTx.txStatus
+      } else {
+        activeTxStatus = .error
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift:57]: Is this displayed in the UI? Concern is that it would show as Tx Confirmation is dismissed when transactions count is 0, which would fall into this case

File: Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift [hunk 3]
```diff
@@ -30,4 +30,27 @@ extension BraveWalletTxService {
       }
     )
   }
+  
+  // Fetches all transactions for all given keyrings
+  func allTransactions(
+    for keyrings: [BraveWallet.KeyringInfo]
+  ) async -> [BraveWallet.TransactionInfo] {
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift]: Can we merge this with the above `pendingTransactions` helper function? We could keep `allTransactions` name but pass in an optional transaction status to filter for

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+struct TransactionStatusView: View {
+  @ObservedObject var confirmationStore: TransactionConfirmationStore
+  let networkStore: NetworkStore
+  let keyringStore: KeyringStore
+  @Binding var isShowingGas: Bool
+  @Binding var isShowingAdvancedSettings: Bool
+  @Binding var transactionDetails: TransactionDetailsStore?
+ 
+  let onClose: () -> Void
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: nit: we use onDismiss throughout wallet

File: Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+          .fontWeight(.semibold)
+        Button(action: confirmationStore.nextTransaction) {
+          Text(Strings.Wallet.next)
+            .fontWeight(.semibold)
+            .foregroundColor(Color(.braveBlurpleTint))
+        }
+      }
+    } else {
+      EmptyView()
+    }
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift]: I think this is from before but since it's an `@ViewBuilder` we shouldn't need this else block 🙂

File: Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+        }
+      },
+      alignment: .bottom
+    )
+  }
+}
+
+/// We needed a `TextEditor` that couldn't be edited and had a clear background color
+/// so we have to fallback to UIKit for this
+struct StaticTextView: UIViewRepresentable {
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift]: Maybe a good time to finally move this to it's own file as it's used in a couple views now

**Review by StephenHeaps — COMMENTED** [review]
A couple initial comments

File: Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift (hunk 1 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift]: fixed in [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)

File: Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift (hunk 2 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift:57]: added failed state (haven't heard back from james with his feedback with the design changes we came up with. but will address next week if there is any) [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 4 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: fixed in [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift (hunk 5 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift]: fixed in [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)

File: Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift (hunk 3 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift]: moved to `TransactionConfirmationStore` as a private extension since that is the class actually uses it. [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift (hunk 6 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift]: updated in [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)

File: Sources/BraveWallet/Crypto/CryptoView.swift [hunk 7]
```diff
@@ -313,6 +313,7 @@ private struct CryptoContainerView<DismissContent: ToolbarContent>: View {
               toolbarDismissContent: toolbarDismissContent,
               onDismiss: {
                 cryptoStore.isPresentingPendingRequest = false
+                cryptoStore.prepare()
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/CryptoView.swift]: Is this change needed? `prepare()` gets called in `BraveWalletTxServiceObserver`, this would add the case where user manually dismisses the pending request which might not be intended

File: Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift (hunk 3 above)
Comment (StephenHeaps) [reply Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift]: The concern was more this helper function was almost direct copy/paste of the above `pendingTransactions` but with the transaction status filter removed, I don't think it makes sense to have two separate helper functions for this. Surely we can re-use code here?
Even changing `pendingTransactions` to use this would help I think:
```
  // Fetches all pending transactions for all given keyrings
  func pendingTransactions(
    for keyrings: [BraveWallet.KeyringInfo]
  ) async -> [BraveWallet.TransactionInfo] {
    await allTransactions(for: keyrings).filter { $0.txStatus == .unapproved }
  }
```

File: Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift (hunk 2 above)
Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift:56]: The error state gets shown during dismiss of `eth_signTransaction`
Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2023-02-20 at 17 10 43

Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift:57]: The error state gets shown during dismiss of `eth_signTransaction` (slow animations toggle used to show it easier)

https://user-images.githubusercontent.com/5314553/220373610-1814915a-5857-456f-a334-b346d3ce088d.mp4

**Comment by StephenHeaps** [comment]
There's a bug if you have a pending request that is lower priority than `eth_signTransaction` that causes the modal to present again but showing the error state:

https://user-images.githubusercontent.com/5314553/220378322-7667bc32-24f8-4ce2-a0de-62d1d11f8515.mp4

STR:
1. create non-transaction dapp request (ex. `personal_sign`)
2. dismiss / minimize the request so it's left pending
3. create `eth_signTransaction` tx
4. confirm the `eth_signTransaction`
5. cancel/sign the dapp request from step 1

We may need a special check when `pendingRequest` contains empty transactions (ex after confirming `eth_signTransaction` here) to verify the completion state is shown (submitted/completed/signed/error views) before we show the dapp request?

File: Sources/BraveWallet/Crypto/CryptoView.swift (hunk 7 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/CryptoView.swift]: it's needed since the `pendingRequest` which was `.transaction([])` need to be set to `nil`

File: Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift (hunk 2 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift:57]: fixed in [cf0e2bb](https://github.com/brave/brave-ios/pull/6960/commits/cf0e2bb683b9aac63b9ea363fdca4e7c43eb009b)

File: Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift (hunk 3 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift]: it's ok to keep the `allTransactions` private for now since i dont think there is any other place uses it. can reconsider if we have more cases that need this helper function.

Comment (StephenHeaps) [reply Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift]: Access control shouldn't prevent us from re-using code here. I don't think we should have duplicated code like this, it's an easy fix here rather than updating 2 spots later if APIs change.

**Comment by StephenHeaps** [comment]
Some state issues if the transaction modal is dismissed when showing the transaction status with the latest changes, previous commit it would re-open and show the error state... In earlier testing of this PR would show the transaction status as it was previously shown 🤔.

https://user-images.githubusercontent.com/5314553/220425203-102f8b9b-48f5-408e-bdee-6f838c61e892.mp4

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+  @Binding var transactionDetails: TransactionDetailsStore?
+ 
+  let onDismiss: () -> Void
+  
+  @Environment(\.openWalletURLAction) private var openWalletURL
+  
+  @ViewBuilder private var loadingTxView: some View {
+    VStack {
+      TxProgressView()
+    }
```
Comment (kylehickinson) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: Doesn't need to be in a `VStack`

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+  @State private var isAnimating = false
+  
+  var body: some View {
+    Image("tx-loading", bundle: .module)
+      .rotationEffect(.degrees(isAnimating ? 360 : 0))
+      .animation(isAnimating ? .linear(duration: 1.0).repeatForever(autoreverses: false) : .default)
+      .onAppear { isAnimating = true }
+      .onDisappear { isAnimating = false }
+  }
+}
```
Comment (kylehickinson) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: Can we not just use `BraveCircularProgressViewStyle`?

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+  
+  var body: some View {
+    if confirmationStore.isTxSubmitting {
+      loadingTxView
+    } else {
+      switch confirmationStore.activeTxStatus {
+      case .signed, .submitted:
+        signedOrSubmittedTxView
+      case .confirmed, .error:
+        if confirmationStore.activeTxStatus == .error, confirmationStore.activeTransactionId == "" {
```
Comment (kylehickinson) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: `confirmationStore.activeTransactionId.isEmpty`

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+                Text(Strings.Wallet.confirmedTransactionReceiptButtonTitle)
+              }
+              .buttonStyle(BraveOutlineButtonStyle(size: .large))
+            }
+            Button {
+              onDismiss()
+            } label: {
+              Text(Strings.Wallet.confirmedTransactionCloseButtonTitle)
+            }
+            .buttonStyle(BraveFilledButtonStyle(size: .large))
```
Comment (kylehickinson) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift:106]: You can move this `buttonStyle` (and the one above) to the `HStack`

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+          }
+          .padding(.top, 40)
+        }
+        .frame(maxWidth: .infinity)
+        .frame(minHeight: geometry.size.height)
```
Comment (kylehickinson) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: These can be the same modifier `.frame(maxWidth: .infinity, minHeight: geometry.size.height)`

File: Sources/BraveWallet/Crypto/Transaction Confirmations/StaticTextView.swift [hunk 13]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
```
Comment (kylehickinson) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/StaticTextView.swift]: Extra whitespace

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 11 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift:106]: they are different button styles tho

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 8 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: Using `ProgressView` with `.braveCircular` style. Move this entirely out of `TransactionStatusView` at [02e39dd](https://github.com/brave/brave-ios/pull/6960/commits/02e39dd3500416bf27c899accce6252b59e78d3f)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 10 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: updated in [02e39dd](https://github.com/brave/brave-ios/pull/6960/commits/02e39dd3500416bf27c899accce6252b59e78d3f)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 9 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: updated in [02e39dd](https://github.com/brave/brave-ios/pull/6960/commits/02e39dd3500416bf27c899accce6252b59e78d3f)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 12 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: updated in [02e39dd](https://github.com/brave/brave-ios/pull/6960/commits/02e39dd3500416bf27c899accce6252b59e78d3f)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 11 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift:110]: updated in [02e39dd](https://github.com/brave/brave-ios/pull/6960/commits/02e39dd3500416bf27c899accce6252b59e78d3f)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/StaticTextView.swift (hunk 13 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/StaticTextView.swift]: update in [02e39dd](https://github.com/brave/brave-ios/pull/6960/commits/02e39dd3500416bf27c899accce6252b59e78d3f)

File: Sources/BraveWallet/Crypto/Stores/CryptoStore.swift [hunk 14]
```diff
@@ -66,6 +66,18 @@ public class CryptoStore: ObservableObject {
...
+         to be dismissed. It could be not ready as there is no pending tx
+         but an active tx is being shown its different state like
+         loading, submitted, completed or failed.
+         As such we need to continue displaying Tx Confirmation until the
+         user taps Ok/Close on the status overlay.
+         */
+        if let confirmationStore = self.confirmationStore, !confirmationStore.isReadyToBeDismissed {
+          pendingRequest = .transactions([])
+          return
+        }
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Stores/CryptoStore.swift]: Should this logic just exist in `prepare()`? We're checking if we're ready to dismiss in `prepare()` and if we're not ready we're assigning `pendingRequest` to nil, then in this didSet we're checking the same conditions again and assigning `pendingRequest` to `.transaction([])`?

Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Stores/CryptoStore.swift]: updated in [002a14b](https://github.com/brave/brave-ios/pull/6960/commits/002a14be0a933cc81482f71a929845faae6ef3b5)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 11 above)
Comment (kylehickinson) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift:106]: Oh sorry missed that!

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionConfirmationView.swift [hunk 15]
```diff
@@ -63,16 +63,26 @@ struct TransactionConfirmationView: View {
...
-            }
-            // update activeTransactionId
-            confirmationStore.updateActiveTxIdAfterSignedClosed()
+        if confirmationStore.isTxSubmitting {
+          ZStack {
+            Rectangle()
+              .fill(Color(.braveGroupedBackground))
+              .edgesIgnoringSafeArea(.all)
+            ProgressView()
+              .progressViewStyle(.braveCircular(size: .large, tint: .braveBlurpleTint))
```
Comment (kylehickinson) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionConfirmationView.swift]: Just use `ProgressView` + `frame` + `background` instead of `ZStack`

```swift
ProgressView()
  .progressViewStyle(.braveCircular(size: .large, tint: .braveBlurpleTint))
  .frame(maxWidth: .infinity, maxHeight: .infinity)
  .background(Color(.braveGroupedBackground).ignoresSafeArea())
```

Comment (kylehickinson) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionConfirmationView.swift]: Also just FYI `edgesIgnoringSafeArea` is deprecated and we should switch to using `ignoresSafeArea` in new code

File: Sources/BraveWallet/Crypto/Stores/CryptoStore.swift [hunk 16]
```diff
@@ -297,12 +297,28 @@ public class CryptoStore: ObservableObject {
...
+           loading, submitted, completed or failed.
+           As such we need to continue displaying Tx Confirmation until the
+           user taps Ok/Close on the status overlay.
+           */
+          newPendingRequest = .transactions([])
+        } else {
+          // check for webpage requests
+          newPendingRequest = await fetchPendingWebpageRequest()
+        }
       }
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Stores/CryptoStore.swift]: nit, this doesn't need nested:
```suggestion
      } else if let store = confirmationStore, !store.isReadyToBeDismissed {
        /*
         We need to check if Tx Confirmation modal is ready
         to be dismissed. It could be not ready as there is no pending tx
         but an active tx is being shown its different state like
         loading, submitted, completed or failed.
         As such we need to continue displaying Tx Confirmation until the
         user taps Ok/Close on the status overlay.
         */
        newPendingRequest = .transactions([])
      } else {
        // check for webpage requests
        newPendingRequest = await fetchPendingWebpageRequest()
      }
```

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+          Text(confirmationStore.activeTxStatus == .signed ? Strings.Wallet.signedTransactionDescription : Strings.Wallet.submittedTransactionDescription)
+            .font(.subheadline)
+            .foregroundColor(Color(.secondaryBraveLabel))
+            .multilineTextAlignment(.center)
+          Button {
+            onDismiss()
+          } label: {
+            Text(Strings.OKString)
+              .padding(.horizontal, 8)
+          }
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: nit:
```suggestion
          Button(action: onDismiss) {
            Text(Strings.OKString)
              .padding(.horizontal, 8)
          }
```

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift [hunk 18]
```diff
@@ -0,0 +1,151 @@
...
+              } label: {
+                Text(Strings.Wallet.confirmedTransactionReceiptButtonTitle)
+              }
+              .buttonStyle(BraveOutlineButtonStyle(size: .large))
+            }
+            Button {
+              onDismiss()
+            } label: {
+              Text(Strings.Wallet.confirmedTransactionCloseButtonTitle)
+            }
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: nit:
```suggestion
            Button(action: onDismiss) {
              Text(Strings.Wallet.confirmedTransactionCloseButtonTitle)
            }
```

**Review by StephenHeaps — APPROVED** [review]
Lgtm, nice work on getting through bugs with the changes here 🎉

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionConfirmationView.swift (hunk 15 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionConfirmationView.swift]: updated in [8d59b95](https://github.com/brave/brave-ios/pull/6960/commits/8d59b95439fdfcd0da5f733c63d51fea80b2a899)

File: Sources/BraveWallet/Crypto/Stores/CryptoStore.swift (hunk 16 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Stores/CryptoStore.swift]: updated in [8d59b95](https://github.com/brave/brave-ios/pull/6960/commits/8d59b95439fdfcd0da5f733c63d51fea80b2a899)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 17 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: updated in [8d59b95](https://github.com/brave/brave-ios/pull/6960/commits/8d59b95439fdfcd0da5f733c63d51fea80b2a899)

File: Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift (hunk 18 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift]: updated in [8d59b95](https://github.com/brave/brave-ios/pull/6960/commits/8d59b95439fdfcd0da5f733c63d51fea80b2a899)

**Review by kylehickinson — APPROVED** [review]
